allauth/account/views.py

```diff
@@ -279,6 +290,15 @@ def get(self, *args, **kwargs):
     def post(self, *args, **kwargs):
         self.object = confirmation = self.get_object()
         confirmation.confirm(self.request)
+
+        # In the event someone clicks on an email confirmation link
+        # for one account while logged into another account,
+        # logout of the currently logged in account.
+        cur_user = self.request.user
+        if cur_user.is_authenticated and \
+                cur_user != confirmation.email_address.user:
```

Comment:
Can you turn this into `if cur_user.pk != confirmation.email_address.user_id` -- that avoids potentially dereferencing a foreign key and avoids the check on `is_authenticated.

Reply:
The logout should only be called if a user is already logged in. Using pk != user_id matches on other occasions. Trying to change this causes account.tests.ConfirmationViewTests to fail due to session variables being deleted. I need to at least check that cur_user.pk is true.

Reply:
@pennersr Your thoughts based on what iarp said?

Reply:
I am not sure I understand the response. My main point is that doing `confirmation.email_address.user` needlessly invokes a DB query to fetch the user -- comparison on `email_address.user_id` will avoid that.

Reply:
Ok I understand what you're talking about now. Your original comment removed the is_authenticated check which breaks logic used in views tested via ConfirmationViewTests. The is_authenticated check is necessary as this logic should only apply when someone is already logged in.